tests/bsim/bluetooth/host/l2cap/stress/src/main.c

```diff
@@ -163,7 +136,19 @@ int recv_cb(struct bt_l2cap_chan *chan, struct net_buf *buf)
 	rx_cnt++;
 
 	/* Verify SDU data matches TX'd data. */
-	ASSERT(memcmp(buf->data, tx_data, buf->len) == 0, "RX data doesn't match TX");
+	int pos = memcmp(buf->data, tx_data, buf->len);
+
+	if (pos != 0) {
+		LOG_ERR("RX data doesn't match TX: pos %d", pos);
+		LOG_HEXDUMP_ERR(buf->data, buf->len, "RX data");
+		LOG_HEXDUMP_INF(tx_data, buf->len, "TX data");
+
+		for (int p = 0; p < buf->len; p++) {
```

Comment:
Suggest to use same type for loop variable as the value compared to (`buf->len`)